src/test/java/com/training/services/tests/LoyalityUpdateTest.java

```diff
@@ -74,4 +75,99 @@ public void testLoyality() {
+     */
+    @Test
+    public void testLoyalityGANegativeWrongAppKey() {
+        headerMap.put(map.get("AppKeyHeader"), map.get("AppKeyWrongValue"));
+
+        com.training.services.ga.loyalty.Response loyaltyNegativeResponse
+                = new RestEngine().getResponsePut(map.get("url_base") + "/v1/guestAccounts/loyalty",
+                headerMap, new Gson().toJson(requestBody))
+                .as(com.training.services.ga.loyalty.Response.class);
+
+        Assertions.assertThat(loyaltyNegativeResponse.getStatus()).isEqualTo(401)
+                .as("Json response status is not 401");
+
+        softAssert.assertThat(loyaltyNegativeResponse.getError().getErrorCode()).isEqualTo("COMMONS-0001")
+                .as("Json response error, errorcode is not COMMONS-0001");
+
+        softAssert.assertThat(loyaltyNegativeResponse.getErrors().get(0).getInternalMessage())
+                .isEqualTo("The API key header is required and should be valid.")
+                .as("The internal message is not equal to, The API key header is required and should be valid.");
+    }
+
+    /**
+     * Putting wrong vds id, In loyality API, and checking if we are getting correct response from sever
+     */
+    @Test
+    public void testLoyalityGANegativeWrongVdsId() {
+        requestBody.setVdsId(authenticationResponse.getPayload().getAccountId() + "777");
+
+        com.training.services.ga.loyalty.Response loyaltyNegativeResponse
+                = new RestEngine().getResponsePut(map.get("url_base") + "/v1/guestAccounts/loyalty",
+                headerMap, new Gson().toJson(requestBody))
+                .as(com.training.services.ga.loyalty.Response.class);
+
+        Assertions.assertThat(loyaltyNegativeResponse.getStatus()).isEqualTo(401)
+                .as("Json response status is not 401");
+
+        softAssert.assertThat(loyaltyNegativeResponse.getErrors().get(0).getInternalMessage())
+                .isEqualTo("The access token or account ID specified in the request appears to be invalid or expired.")
+                .as("The internal message is not equal to, The access token or account ID specified in the request appears to be invalid or expired.");
+
+        softAssert.assertThat(loyaltyNegativeResponse.getErrors().get(0).getUserMessage())
+                .isEqualTo("The user session appears to be invalid or has expired.")
+                .as("The internal message is not equal to, The user session appears to be invalid or has expired.");
+    }
+
+    /**
+     * Putting wrong access key, In loyality API, and checking if we are getting correct response from sever
+     */
+    @Test
+    public void testLoyalityGANegativeWrongAccessKey() {
+        headerMap.put(map.get("accessToken"), authenticationResponse.getPayload().getAccessToken() + "7777");
+
+        com.training.services.ga.loyalty.Response loyaltyNegativeResponse
+                = new RestEngine().getResponsePut(map.get("url_base") + "/v1/guestAccounts/loyalty",
```

Comment:
can we create a global variable url = map.get("url_base") + "/v1/guestAccounts/loyalty"
and use it all our tests?